srcs/validator.cc

```diff
@@ -0,0 +1,40 @@
+#include <queue>
+#include <vector>
+#include <fruit/fruit.h>
+#include "exception.hpp"
+#include "token.hpp"
+#include "validator.hpp"
+
+class ParaValidator: public Validator{
+    public:
+    using Inject = ParaValidator();
+    ParaValidator() = default;
```

Comment:
Is this required?

Reply:
Can discuss this offline a bit more on whether validator is unnecessary. As for the purpose of this PR, I will keep the validator as is.

Reply:
Sorry, I meant is `ParaValidator() = default` required.  What's your code policy on default constructors?

Reply:
Removed.